include/nanorange/view/subrange.hpp

```diff
@@ -43,14 +43,47 @@ using subrange_::subrange;
 
 namespace detail {
 
+// libstdc++ < 7 does not have a SFINAE-friendly std::tuple_size,
+// meaning that doing virtually anything with it is a hard error, including
+// testing the PairLike concept below. As a workaround, we'll define our own
+// tuple_size (only for std::tuple and std::pair) if we're using libstdc++ v5 or
+// v6.
+//
+// FIXME: Do this better.
+//
+// Note:
+// __GNUC__ tells us we're using GCC or Clang
+// !_LIBCPP_VERSION tells us we're not using libc++
+// !_GLIBCXX_RELEASE tells us we're not using libstdc++ >= 7, which is when this
+// symbol was added
+#if defined(__GNUC__) && !defined(_LIBCPP_VERSION) && !defined(_GLIBCXX_RELEASE)
+template <typename, typename = void>
```

Comment:
The second template parameter is not used by any of the partial specializations.